br/pkg/lightning/mydump/csv_parser.go

```diff
@@ -336,6 +337,9 @@ func (parser *CSVParser) readUntil(chars *byteSet) ([]byte, byte, error) {
 	var buf []byte
 	for {
 		buf = append(buf, parser.buf...)
+		if len(buf) > tidbconfig.MaxTxnEntrySizeLimit {
```

Comment:
Prefer to do this check in `readRecord`.

Reply:
But `readRecord` will invoke `readUntil`. If we check it in `readRecord` and there is no targeted byte in the following content, then too much memory will still be allocated for `buf`.

```
func (parser *CSVParser) readRecord(dst []string) ([]string, error) {
	parser.recordBuffer = parser.recordBuffer[:0]
	parser.fieldIndexes = parser.fieldIndexes[:0]

	isEmptyLine := true
	whitespaceLine := true
	prevToken := csvTokenNewLine
	var firstToken csvToken

outside:
	for {
		content, firstByte, err := parser.readUntil(&parser.unquoteByteSet)

```

Reply:
Oh, I missed it.